### PR TITLE
Added specific field and map search button

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Search/MultiSearch.php [hunk 1]
```diff
@@ -32,6 +32,31 @@ class MultiSearch
...
+
+    /**
+     * Elastic index mapping for theme keywords.
+     */
+    const ELASTIC_INDEX_MAPPING_THEME_KEYWORDS = 'datasetSubmission.themeKeywords';
+
+    /**
+     * Index boost for Title, Authors, Theme Keywords.
+     */
+    const BOOST = '^2';
```
Comment (mickel1138) [inline src/Search/MultiSearch.php:58]: Make private? Also, other maybe public?

File: src/Search/SearchOptions.php [hunk 2]
```diff
@@ -119,6 +119,13 @@ class SearchOptions
      */
     private $rangeEndDate;
 
+    /**
+     * Specific field to be searched upon.
+     *
+     * @var string
+     */
+    private $field;
```
Comment (mickel1138) [inline src/Search/SearchOptions.php:127]: Shall we start type hinting here? I hear `@var` is a bad habit.

**Review by fightingtexasaggie — COMMENTED** [review]
Search for 'all' doesn't seem to work right. Example: searching for 'dolphin' in title returns 19 results, but no results when set to 'all'.

File: src/Controller/UI/MultiSearchController.php [hunk 3]
```diff
@@ -54,6 +55,7 @@ public function search(Request $request, MultiSearch $multiSearch, JsonSerialize
         $searchOptions->setRangeStartDate($rangeStartDate);
         $searchOptions->setRangeEndDate($rangeEndDate);
         $searchOptions->setSortOrder($sortOrder);
+        $searchOptions->setField($field);
```
Comment (fightingtexasaggie) [inline src/Controller/UI/MultiSearchController.php:58]: perhaps only call if not empty string

**Review by fightingtexasaggie — APPROVED** [review]
Confirmed it is now working as expected.

**Pull request merged**